src/Microsoft.OData.Core/ODataContextUrlInfo.cs

```diff
@@ -407,8 +413,13 @@ private static string ProcessSubExpand(string expandNode, string subExpand)
         private static string CombineSelectAndExpandResult(IList<string> selectList, IList<string> expandList)
         {
             string currentExpandClause = string.Empty;
+
             if (selectList.Any())
             {
+                // https://github.com/OData/odata.net/issues/1104
+                // If the user explicitly selects and expands a nav prop, we should include both forms in contextUrl
+                // We can't, though, because SelectExpandBinder.AddExplicitNavPropsWhereNeeded adds all of the expanded items

```

Comment:
>SelectExpandBinder.AddExplicitNavPropsWhereNeeded [](start = 45, length = 49)

This method doesn't exist. Not mentioned in PR #1104 either. Should it be SelectExpandClauseFinisher.AddExplicitNavPropLinksWhereNecessary instead?

Reply:
Yes; thanks.

---
In reply to: [175302307](https://github.com/OData/odata.net/pull/1109#discussion_r175302307) [](ancestors = 175302307)